src/Admin/AbstractAdmin.php

```diff
@@ -811,6 +811,8 @@ public function buildDatagrid()
         }
 
         $filterParameters = $this->getFilterParameters();
+        $sortingParameters = $this->determineSorting();
+        $filterParameters = array_merge($filterParameters, $sortingParameters);
```

Comment:
This way you could remove [the assignment at line #813](https://github.com/sonata-project/SonataAdminBundle/pull/5598/files#diff-61e6d1f8fafe87b868adbe28ce0ba8f6R813):
```suggestion
        $filterParameters = array_merge($filterParameters, $this->getFilterParameters());
```

Reply:
I've done suggested modifications. Take a look at 91eb73e5 commit, please.